pkg/agent/controller/networkpolicy/reject_test.go

```diff
@@ -502,3 +522,244 @@ func Test_getRejectPacketOutMutateFunc(t *testing.T) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeDIface, true)
+				mockClient.EXPECT().SendICMPPacketOut(fakeMac2, fakeMac1, fakeIPv42, fakeIPv41, gomock.Any(), gomock.Any(), false, uint8(3), uint8(10), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+		{
+			name: "IPv4,TCP,antreaProxyDisable,podLocal",
+			args: args{
+				srcMAC:            fakeMac1,
+				dstMAC:            fakeMac2,
+				srcIP:             fakeIPv41,
+				dstIP:             fakeIPv42,
+				isIPv6:            false,
+				isTCP:             true,
+				srcPort:           fakePort1,
+				dstPort:           fakePort2,
+				tcpSeqNum:         fakeTCPSeqNum,
+				antreaProxyEnable: false,
+			},
+			expectFunc: func(mockClient *testing2.MockClient, mockIStore *testing3.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeSIface, true)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeDIface, true)
+				mockIStore.EXPECT().GetInterfacesByType(gomock.Any()).Return([]*interfacestore.InterfaceConfig{fakeGWIface})
+				mockClient.EXPECT().SendTCPPacketOut(fakeMac2, fakeMac1, fakeIPv42, fakeIPv41, gomock.Any(), gomock.Any(), false, fakePort2, fakePort1, gomock.Any(), fakeTCPSeqNum+1, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+		{
+			name: "IPv6,TCP,antreaProxy,podLocal",
+			args: args{
+				srcMAC:            fakeMac1,
+				dstMAC:            fakeMac2,
+				srcIP:             fakeIPv61,
+				dstIP:             fakeIPv62,
+				isIPv6:            true,
+				isTCP:             true,
+				srcPort:           fakePort1,
+				dstPort:           fakePort2,
+				tcpSeqNum:         fakeTCPSeqNum,
+				antreaProxyEnable: true,
+			},
+			expectFunc: func(mockClient *testing2.MockClient, mockIStore *testing3.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeSIface, true)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeDIface, true)
+				mockClient.EXPECT().SendTCPPacketOut(fakeMac2, fakeMac1, fakeIPv62, fakeIPv61, gomock.Any(), gomock.Any(), true, fakePort2, fakePort1, gomock.Any(), fakeTCPSeqNum+1, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+	}
+	ctrl := gomock.NewController(t)
+	controller, _, _ := newTestController()
+	mockClient := testing2.NewMockClient(ctrl)
+	mockIStore := testing3.NewMockInterfaceStore(ctrl)
+	controller.ofClient = mockClient
+	controller.ifaceStore = mockIStore
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			controller.antreaProxyEnabled = tt.args.antreaProxyEnable
+			tt.expectFunc(mockClient, mockIStore)
+			assert.NoError(t, controller.rejectRequest(genPacketIn(tt.args.srcMAC, tt.args.dstMAC, tt.args.srcIP, tt.args.dstIP, tt.args.srcPort, tt.args.dstPort, tt.args.tcpSeqNum, tt.args.isIPv6, tt.args.isTCP, tt.args.matches)))
```

Comment:
It's a quite long args list, I think you can define a `*ofctrl.PacketIn` field in `tests` struct and call `genPacketIn` directly to get the PacketIn.